litex_boards/platforms/zedboard.py

```diff
@@ -0,0 +1,256 @@
+        # "GBTCLK0_M2C_P": "",  # NC
+        "CLK0_M2C_N": "L19",
+        "CLK0_M2C_P": "L18",
+        "CLK1_M2C_N": "C19",
+        "CLK1_M2C_P": "D18",
+        "IIC_SCL_MAIN": "R7",
+        "IIC_SDA_MAIN": "U7",
+        "LA00_CC_N": "M20",
+        "LA00_CC_P": "M19",
+        "LA01_CC_N": "N20",
+        "LA01_CC_P": "N19",
+        "LA02_N": "P18",
+        "LA02_P": "P17",
+        "LA03_N": "P22",
+        "LA03_P": "N22",
+        "LA04_N": "M22",
+        "LA04_P": "M21",
+        "LA05_N": "K18",
+        "LA05_P": "J18",
+        "LA06_N": "L22",
+        "LA06_P": "L21",
+        "LA07_N": "T17",
+        "LA07_P": "T16",
+        "LA08_N": "J22",
+        "LA08_P": "J21",
+        "LA09_N": "R21",
+        "LA09_P": "R20",
+        "LA10_N": "T19",
+        "LA10_P": "R19",
+        "LA11_N": "N18",
+        "LA11_P": "N17",
+        "LA12_N": "P21",
+        "LA12_P": "P20",
+        "LA13_N": "M17",
+        "LA13_P": "L17",
+        "LA14_N": "K20",
+        "LA14_P": "K19",
+        "LA15_N": "J17",
+        "LA15_P": "J16",
+        "LA16_N": "K21",
+        "LA16_P": "J20",
+        "LA17_CC_N": "B20",
+        "LA17_CC_P": "B19",
+        "LA18_CC_N": "C20",
+        "LA18_CC_P": "D20",
+        "LA19_N": "G16",
+        "LA19_P": "G15",
+        "LA20_N": "G21",
+        "LA20_P": "G20",
+        "LA21_N": "E20",
+        "LA21_P": "E19",
+        "LA22_N": "F19",
+        "LA22_P": "G19",
+        "LA23_N": "D15",
+        "LA23_P": "E15",
+        "LA24_N": "A19",
+        "LA24_P": "A18",
+        "LA25_N": "C22",
+        "LA25_P": "D22",
+        "LA26_N": "E18",
+        "LA26_P": "F18",
+        "LA27_N": "D21",
+        "LA27_P": "E21",
+        "LA28_N": "A17",
+        "LA28_P": "A16",
+        "LA29_N": "C18",
+        "LA29_P": "C17",
+        "LA30_N": "B15",
+        "LA30_P": "C15",
+        "LA31_N": "B17",
+        "LA31_P": "B16",
+        "LA32_N": "A22",
+        "LA32_P": "A21",
+        "LA33_N": "B22",
+        "LA33_P": "B21",
+        "PRSNT_M2C_L": "AB14"
+    })
+]
+
+
+class Platform(XilinxPlatform):
+    default_clk_name = "clk100"
+    default_clk_period = 10.0
+
+    def __init__(self, toolchain="vivado"):
+        XilinxPlatform.__init__(self, "xc7z020clg484-1", _io, _connectors,
+                                toolchain=toolchain)
+        # self.toolchain.bitstream_commands = \
+        #     ["set_property BITSTREAM.CONFIG.SPI_BUSWIDTH 4 [current_design]"]
+        # self.toolchain.additional_commands = \
+        #     ["write_cfgmem -force -format bin -interface spix4 -size 16 "
+        #      "-loadbit \"up 0x0 {build_name}.bit\" -file {build_name}.bin"]
+        # self.toolchain.additional_commands = [
+        #     # Generate .bit.bin file for loading with linux fpga_manager
+        #     'exec bootgen -image {build_name}.bif -w -arch zynq -process_bitstream bin'
+        # ]
+
+    def add_oled(self, soc, SPI_N=0, SS_N=1, DC_GPIO=8, RST_GPIO=9):
```

Comment:
We generally don't include logic/integration code directly in the platform, can you move this to the target?